lib/commands/find.js

```diff
@@ -34,23 +35,43 @@ commands.findElOrEls = async function (strategy, selector, mult, context) {
     return JSON.stringify(commandObject);
   };
 
-  let findByAxIdCmd = createGetElementCommand(strategy, selector, mult, context);
+  let result;
+  let doFind = async () => {
+    let findByAxIdCmd = createGetElementCommand(strategy, selector, mult, context);
+  
+    let res = await this.executeSocketCommand(findByAxIdCmd);
 
-  let data = await this.executeSocketCommand(findByAxIdCmd);
+    try {
+      result = JSON.parse(res);
+    } catch (e) {
+      // parse error
+      throw new Error("Bad response from findElOrEls");
+    }
+
+    // check the status or resulting value to see that it's not empty
+    if (result.value === "")
+    {
+      return false;
+    }
+    return true;
+  };
 
-  let result;
   try {
-    result = JSON.parse(data);
-  } catch (e) {
-    throw new Error("Bad response from findElOrEls");
+    await this.implicitWaitForCondition(doFind);    
+  } catch (err) {
+    if (err.message && err.message.match(/Condition unmet/)){
+      // condition was not met setting res to empty array which should trip the NoSuchElementError
+      result = [];
+    } else {
+      // some other issue occurred, report it
+      throw err;
+    }
   }
-
-  // get status returned
-  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_NO_SUCH_ELEMENT)
+  // make sure we found something - if not throw NoSuchElementError
+  if (!result || _.size(result) === 0) {
     throw new errors.NoSuchElementError();
-
-
+  }
```

Comment:
I tweaked this. Now if the conditions are not met whether doing a find or finds, it will throw a element not found error. Tested this with single, multiple for both success and failure with and without the implicit wait.